### PR TITLE
[provider] unit tests for predicate functions

### LINKED ISSUE #772: [provider] unit tests for predicate functions
This issue holds true once this PR https://github.com/envoyproxy/gateway/pull/702 is merged.
All predicate functions mentioned in `predicate.go` are candidates for unit tests. 
Some of them already have unit tests, as they lived even after the refactor.

The PR adds new predicate functions that require additional unit tests.

### PR DESCRIPTION
**Pull request by chauhanshubham**
Resolves #772 
Signed-off-by: Shubham Chauhan <shubham@tetrate.io>

### CODE REVIEW

File: internal/provider/kubernetes/test/utils.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package test
```
Comment (danehans) [inline internal/provider/kubernetes/test/utils.go:6]: None of the types, func, etc. are specific to testing. Why not have this as a `utils`, `resources`, etc. pkg?

Comment (chauhanshubham) [reply internal/provider/kubernetes/test/utils.go:6]: these just return sample objects that are only being used in _tests

File: internal/provider/kubernetes/predicates.go [hunk 2]
```diff
@@ -121,7 +121,7 @@ func (r *gatewayAPIReconciler) validateServiceForReconcile(obj client.Object) bo
 		}
 
 		r.statusUpdateForGateway(gtw, svc, deployment)
-		return true
+		return false
```
Comment (danehans) [inline internal/provider/kubernetes/predicates.go:124]: @chauhanshubham why is `false` now being returned?

File: internal/provider/kubernetes/predicates.go [hunk 3]
```diff
@@ -209,6 +209,7 @@ func (r *gatewayAPIReconciler) envoyServiceForGateway(ctx context.Context, gatew
 		if kerrors.IsNotFound(err) {
 			return nil, nil
 		}
+		// return nil, nil
```
Comment (danehans) [inline internal/provider/kubernetes/predicates.go:212]: Please remove this comment.

File: internal/provider/kubernetes/predicates_test.go [hunk 4]
```diff
@@ -9,11 +9,14 @@ import (
 	"testing"
 
 	"github.com/envoyproxy/gateway/api/config/v1alpha1"
+	"github.com/envoyproxy/gateway/internal/envoygateway"
+	"github.com/envoyproxy/gateway/internal/gatewayapi"
```
Comment (danehans) [inline internal/provider/kubernetes/predicates_test.go:13]: Please reorganize these imports as follows:
- std pkgs
- upstream pkgs
- internal pkgs

**Review by danehans — CHANGES_REQUESTED** [review]

**Review by danehans — APPROVED** [review]

**Pull request merged**